### PR TITLE
Update Parse.hs

### PR DESCRIPTION
**Pull request by odr**
Adding tagAttr - more common `tag` function with checking attributes. [Issue 60.](https://github.com/snoyberg/xml/issues/60)

### CODE REVIEW

File: xml-conduit/Text/XML/Stream/Parse.hs [hunk 1]
```diff
@@ -604,29 +605,30 @@ contentMaybe = do
...
-tag checkName attrParser f = do
+tagAttr check attrParser f = do
     x <- dropWS
     case x of
         Just (EventBeginElement name as) ->
-            case checkName name of
+            case check name as of
                 Just y ->
                     case runAttrParser' (attrParser y) as of
                         Left e -> lift $ monadThrow e
```
Comment (k0ral) [inline xml-conduit/Text/XML/Stream/Parse.hs]: I had a different implementation in mind: what about simply replacing this `monadThrow` with `return Nothing` ?

Also, the new `tagAttr` signature looks unnecessarily redundant: the attributes parsing function is fed twice:
- through `[(Name, [Content])] -> Maybe a`
- through `a -> AttrParser b`

File: xml-conduit/Text/XML/Stream/Parse.hs [hunk 2]
```diff
@@ -604,29 +605,30 @@ contentMaybe = do
...
--- | The most generic way to parse a tag. It takes a predicate for checking if
--- this is the correct tag name, an 'AttrParser' for handling attributes, and
--- then a parser for dealing with content.
+-- | The most generic way to parse a tag. It takes 
+-- a predicate for checking if this is the correct tag name and attributes, 
+-- an 'AttrParser' for handling attributes, 
+-- and then a parser for dealing with content.
 --
 -- This function automatically absorbs its balancing closing tag, and will
 -- throw an exception if not all of the attributes or child elements are
```
Comment (k0ral) [inline xml-conduit/Text/XML/Stream/Parse.hs]: This comment should be updated to describe the new behavior with respect to attributes.

Comment (odr) [inline xml-conduit/Text/XML/Stream/Parse.hs]: FYI I am away for two week

**Comment by k0ral** [comment]
Implemented in #93.

**Pull request closed without merging**